sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -103,6 +116,135 @@ class FileScanRDD(
         context.killTaskIfInterrupted()
         (currentIterator != null && currentIterator.hasNext) || nextIterator()
       }
+
+      ///////////////////////////
+      // FILE METADATA METHODS //
+      ///////////////////////////
+
+      // whether a metadata column exists and it is a `MetadataAttribute`
+      private lazy val hasMetadataAttribute: Boolean = {
```

Comment:
do we need this `lazy val`? I think this is simply `metadataStructCol.isDefined`, as the caller side should always pass in the metadata col.

Reply:
ok, just want to be super safe, but yea, make sense! thanks